api/base.proto

```diff
@@ -157,6 +157,22 @@ message HeaderValueOption {
   google.protobuf.BoolValue append = 2;
 }
 
+// Data source consisting of either a file or an inline value.
+message DataSource {
+  oneof specifier {
+    option (validate.required) = true;
+
+    // Local filesystem data source.
+    string filename = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Bytes inlined in the configuration.
+    bytes inline = 2 [(validate.rules).bytes.min_len = 1];
```

Comment:
Name this `inline_bytes`. We don't have a backwards compatibility requirement for field names; in general we avoid churn, but it's fine to do this where it is a win.